src/lib/message/L1Transaction.ts

```diff
@@ -217,10 +217,18 @@ export class L1TransactionReceipt implements TransactionReceipt {
     }
   }
 
+  public async getAllDeposits<T extends SignerOrProvider>(l2SignerOrProvider: T) {
```

Comment:
I don't think this adds much value, a caller can do this themselves if they want both. 

EthDepositMessage and L1ToL2Message are quite different objects, so in most cases a caller will need to be filtering these objects anyway, checking the type, then doing different actions for each type.

Reply:
removed